python/fusion_engine_client/parsers/mixed_log_reader.py

```diff
@@ -107,6 +111,18 @@ def __init__(self, input_file, warn_on_gaps: bool = False, show_progress: bool =
         self._original_index = fast_indexer.fast_generate_index(input_path, force_reindex=ignore_index,
                                                                 save_index=save_index, max_bytes=max_bytes)
         self.next_index_elem = 0
+        self.index = self._original_index
+        self.filtered_message_types = False
+
+        # Now, apply filters -> apply pose message filter, then look at available message types
+        # self.populate_xx() (builds list of source_ids) -> self.filter_in_place(source_id) -> applies pose message filtering
+            # get list of available source IDs
+        self.get_available_source_ids()
```

Comment:
You don't define `self.available_source_ids` in the constructor. You need to do that before calling `get_available_source_ids()` -- all member variables should always be declared in the constructor for clarity, even though technically Python lets you do it later.